libvast/test/system/index.cpp

```diff
@@ -199,3 +218,16 @@ TEST(iterable bro conn log query result) {
 }
 
 FIXTURE_SCOPE_END()
+
+FIXTURE_SCOPE(meta_index_setup_test, synopsis_fixture)
+
+TEST(meta index factory) {
+  auto factory = get_synopsis_factory(sys);
+  REQUIRE(factory);
+  CHECK_EQUAL(factory->first, caf::atom("Sy_TEST"));
+  CHECK(factory->second == make_synopsis);
+  // We could additionally go into the state of the actor and verify that the
```

Comment:
We should absolutely verify the INDEX. It's also simple with CAF's testing DSL:

```c++
MESSAGE("run initialization code");
run();
MESSAGE("verify INDEX state");
auto& ref = deref<stateful_actor<index_state>>(index);
CHECK_EQUAL(ref.state. ...);
```

Reply:
Exactly the code piece I was looking for but couldn't find when skimming the code base. Thanks! I've added this snippet as separate commit.